R/tar-terra-rast.R

```diff
@@ -6,6 +6,12 @@
 #'   to [terra::writeRaster()]
 #' @param gdal character. GDAL driver specific datasource creation options
 #'   passed to [terra::writeRaster()]
+#' @param preserve_metadata logical. When `FALSE` (default), any auxiliary files
```

Comment:
Ever since reading https://design.tidyverse.org/boolean-strategies.html (and to some extent, https://design.tidyverse.org/argument-clutter.html ) I now always take a moment to consider TRUE/FALSE arguments as a potential opportunity to use a different set of arguments. E.g., `preserve_metadata` could instead be "drop" or "zip", perhaps? This also helps us keep open other options for dealing with this in the future. 

It does create more work to set up different options, but I think it might be worthwhile, but there are tradeoffs as well. What are your thoughts?

Reply:
I also think it might be worthwhile to allow for different options and use a set of character method names rather than boolean. I can imagine, for instance, future terra or GDAL versions that have better support for /vsizip/ read and write  that users might want to be able to opt-in to.

Reply:
Makes sense.  I'll change the options to `"drop"` and `"zip"` and update the documentation accordingly.  Might be able to add `"tar"` (you can make tar archives without compression, but I can't get base R `tar()` to work) and `"vsizip"` as options in the future.